centinel/experiments/baseline.py

```diff
@@ -0,0 +1,271 @@
+            # self.traceroute_methods = ["icmp", "udp", "tcp"]
+            self.traceroute_methods = ["tcp", "udp"]
+
+    def run(self):
+        if self.record_pcaps:
+            self.external_results = {}
+
+        for input_file in self.input_files.items():
+            logging.info("Testing input file %s..." % (input_file[0]))
+            self.results.append(self.run_file(input_file))
+
+    def run_file(self, input_file):
+        file_name, file_contents = input_file
+
+        # Initialize the results for this input file.
+        # This can be anything from file name to version
+        # to any useful information.
+        result = {}
+        result["file_name"] = file_name
+
+
+        http_results = {}
+        tls_results = {}
+        dns_results = {}
+        traceroute_results = {}
+        url_metadata_results = {}
+        file_metadata = {}
+        file_comments = []
+
+        # each pcap file is stored in a separate file
+        # designated by a number. the indexes are stored
+        # in the json file and the pcap files are stored
+        # with their indexes as file names.
+        pcap_results = {}
+        pcap_indexes = {}
+        url_index = 0
+        comments = ""
+
+        # we may want to make this threaded and concurrent
+        for line in file_contents:
+            line = line.strip()
+
+            # parse file comments, if it looks like "key : value",
+            # parse it as a key-value pair. otherwise, just
+            # store it as a raw comment.
+            if line[0] == '#':
+                line = line[1:].strip()
+                if len(line.split(':')) > 1:
+                    key, value = line.split(':', 1)
+                    key = key.strip()
+                    value = value.strip()
+                    file_metadata[key] = value
+                else:
+                    file_comments.append(line)
+                continue
+
+            url = line
+            meta = ''
+            url_index = url_index + 1
+
+            # handle cases where URL is enclosed in quotes
+            if line[0] == '"':
+                next_quote = line[1:].find('"')
+                url = line[1:next_quote + 1]
+                # try to separate metadata, if any
+                try:
+                    meta = ','.join(line[next_quote + 2:].split(',')[1:])
+                except:
+                    pass
+
+            # if the line doesn't start with a quote
+            # but the list entry has comma separated meta-data
+            elif len(line.split(',')) > 1:
+                url, meta = line.split(',', 1)
+                # remove trailing spaces
+                url = url.strip()
+
+            http_ssl = False
+            ssl_port = 443
+            http_path = '/'
+
+            # parse the URL to extract netlocation, HTTP path, domain name,
+            # and HTTP method (SSL or plain)
+            try:
+                http_netloc = ''.join(urlparse.urlparse(url).netloc)
```

Comment:
you call urlparse multiple times on the same URL. You should do the parse once, store it an a variable, and reference it later.